File: tests/fixtures/go/checksum.go
// Package fixture is the smallest program that still exercises a real Go
// toolchain: it builds, it has a test, a benchmark and a fuzz target, so one
// tree can prove go-ci, benchmark and fuzzing without pulling a dependency.
package fixture

// Checksum sums the bytes of s modulo 256. Trivial on purpose: the point of
// this fixture is to make the workflow run, not to test arithmetic.
func Checksum(s string) int {
	total := 0
	for _, b := range []byte(s) {
		total += int(b)
	}
	return total % 256
}

File: tests/fixtures/r/R/checksum.R
#' Sum the bytes of a string modulo 256.
#'
#' Trivial on purpose: this fixture exists to make the workflow run, not to
#' test arithmetic.
#'
#' @param text A length-one character vector.
#' @return An integer in [0, 255].
#' @export
checksum <- function(text) {
  if (nchar(text) == 0L) {
    return(0L)
  }
  sum(as.integer(charToRaw(text))) %% 256L
}
